app/services/dqt/sync_state.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+module Dqt
+  class SyncState
+    include ServicePattern
+
+    AWARDED = "Pass"
+    WITHDRAWN = "Withdrawn"
+    DEFERRED = "Deferred"
+
+    def initialize(trainee:)
+      @trainee = trainee
+    end
+
+    def call
+      return unless FeatureService.enabled?(:integrate_with_dqt)
+      return unless [AWARDED, WITHDRAWN, DEFERRED].include?(dqt_state)
+
+      Trainees::Update.call(trainee: trainee, params: update_params, update_dqt: false)
+    end
+
+  private
+
+    attr_reader :trainee
+
+    def response
+      @response ||= Dqt::RetrieveTeacher.call(trainee: trainee)
+    end
+
+    def dqt_state
+      @dqt_state ||= response.dig("initial_teacher_training", "result")
+    end
+
+    def update_params
+      case dqt_state
+      when AWARDED
+        awarded_at = response.dig("qualified_teacher_status", "qts_date")
+        { state: "awarded", awarded_at: awarded_at }
```

Comment:
I'm not using `trainee.award_qts!` here because it's actually an invalid transition according to our state machine (`trn_received` => `recommended_for_award` => `awarded`) but I thought no biggie?

Reply:
we've also got a field outcome_date in Reg - should we update that too?

Reply:
eep, i'm not sure.. i don't think we can assume that this awarded_at is the same as outcome_date, so I'm going to err on the side of not saving possibly incorrect data.

Reply:
Doesn't the outcome endpoint on DQT return the date you give it plus 1 day? Maybe we could the opposite? That is hardcoding DQT logic into register though. Maybe we don't want to do that.

Reply:
I have never come across outcome_date so I can't be too useful here. I hadn't realised there was another field in addition to the date when the trainee in reality got QTS, and the date when the provider told us about it. How does the outcome_date field usually work?

To me, it's more important to when when in reality the trainee got QTS than any other datapoint for what it's worth.

Reply:
#whatgrahamsaid - populate outcome date with the award date minus one day.

Reply:
Hmm is it always the next day? could it be the next working day?

Reply:
leaving as nil for now so as not to block this work.